src/components/mesh.tsx

```diff
@@ -0,0 +1,52 @@
+import { JSX, ReactNode, forwardRef, useRef, useState } from 'react';
+import { ThreeEvent } from '@react-three/fiber';
+import * as THREE from 'three';
+
+export type GenericMeshProps = JSX.IntrinsicElements['mesh'] & {
+  material?: ReactNode;
+  parentCallback?: (ref: React.RefObject<THREE.Mesh>) => void;
+  children?: ReactNode;
+};
```

Comment:
w sumie jak nie ma `material` to mozna tutaj uzyc `ControllerProps` i usunac ten caly typ